DFC.App.ContactUs/Views/EnterYourDetails/_CallbackDateTimeEditor.cshtml

```diff
@@ -0,0 +1,132 @@
+﻿@model EnterYourDetailsBodyViewModel
+
+@{
+    const string govukErrorClass = "govuk-form-group--error";
+    var showErrorClass = string.Empty;
+    var showErrorClassForDay = string.Empty;
+    var showErrorClassForMonth = string.Empty;
+    var showErrorClassForYear = string.Empty;
+    var showErrorClassForHour = string.Empty;
+    var showErrorClassForMinute = string.Empty;
+    var nameFieldForDay = Html.NameFor(model => model.CallbackDateTime.Day);
+    var nameFieldForMonth = Html.NameFor(model => model.CallbackDateTime.Month);
+    var nameFieldForYear = Html.NameFor(model => model.CallbackDateTime.Year);
+    var nameFieldForHour = Html.NameFor(model => model.CallbackDateTime.Hour);
+    var nameFieldForMinute = Html.NameFor(model => model.CallbackDateTime.Minute);
+    var errorKeys = new[] { Html.NameFor(model => model.CallbackDateTime), nameFieldForDay, nameFieldForMonth, nameFieldForYear, nameFieldForHour, nameFieldForMinute, };
+
+    foreach (var errorKey in errorKeys)
+    {
+        if (ViewData.ModelState.Keys.Contains(errorKey))
+        {
+            if (ViewData.ModelState[errorKey] != null && ViewData.ModelState[errorKey].Errors.Any())
+            {
+                if (errorKey.Equals(nameFieldForDay, StringComparison.OrdinalIgnoreCase))
```

Comment:
Wouldn't it be cleaner to use switch statement?

Reply:
it would be but I need to use string values and not constants that a switch would require